failed tests/test_model.py::TestUserDefiniedModel::test_composite_plotting

Hi all,

This is my first issue here. Yes, I read the instructions and I am sure this is a GitHub Issue.

I just forked the repo, created a clean conda env, installed the dependencies (using `pip`), and ran `py.test`. I got a single failing test. 

Output of `py.test --lf --tb=short`:
```
tests\test_model.py:792: in test_composite_plotting
    fig, ax = result.plot(show_init=True)
E   TypeError: cannot unpack non-iterable NoneType object
```

Somehow the guarded import at line 39 in `lmfit.model` fails to import `matplotlib`, yet the `pytest.importorskip("matplotlib")` (and ensuing `import matplotlib`) succeeds. (Direct `import matplotlib` statements also work..). The `_ensureMatplotlib` decorator causes `result.plot` to return `None`.

The specific import error around line 39 of `lmfit.model` is:
```
OSError: no library called "cairo" was found
no library called "libcairo-2" was found
cannot load library 'libcairo.so.2': error 0x7e
cannot load library 'libcairo.2.dylib': error 0x7e
cannot load library 'libcairo-2.dll': error 0x7e
```

Running `pip install pycairo` fixes the issue. Adding `pycairo` to the list of requirements would likely fix this issue.

For completeness, here is version info:
```python
>>> import sys, lmfit, numpy, scipy, asteval, uncertainties, six
>>> print('Python: {}\n\nlmfit: {}, scipy: {}, numpy: {}, asteval: {}, uncertainties: {}, six: {}'\
...       .format(sys.version, lmfit.__version__, scipy.__version__, numpy.__version__, \
...       asteval.__version__, uncertainties.__version__, six.__version__))
Python: 3.9.1 | packaged by conda-forge | (default, Jan 26 2021, 01:29:07) [MSC v.1916 64 bit (AMD64)]

lmfit: 1.0.2, scipy: 1.6.1, numpy: 1.20.1, asteval: 0.9.22, uncertainties: 3.1.5, six: 1.15.0
```
I got the same error using the latest commit on `master`.

--- a/tests/test_model.py
+++ b/tests/test_model.py
@@ -764,6 +764,7 @@ class TestUserDefiniedModel(CommonTests, unittest.TestCase):
         for _, par in pars.items():
             assert len(repr(par)) > 5
 
+    @pytest.mark.skipif(not lmfit.model._HAS_MATPLOTLIB, reason="requires matplotlib.pyplot")
     def test_composite_plotting(self):
         # test that a composite model has non-empty best_values
         pytest.importorskip("matplotlib")

